clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java

```diff
@@ -77,6 +77,18 @@
 @PowerMockIgnore("javax.management.*")
 public class KafkaProducerTest {
 
+    @Test
+    public void testMetricsReporterAutoGeneratedClientId() {
+        Properties props = new Properties();
+        props.setProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9999");
+        props.setProperty(ProducerConfig.METRIC_REPORTER_CLASSES_CONFIG, MockMetricsReporter.class.getName());
+        KafkaProducer<String, String> producer = new KafkaProducer<>(
+                props, new StringSerializer(), new StringSerializer());
+
+        Assert.assertEquals(producer.getClientId(), MockMetricsReporter.clientId);
```

Comment:
generally in assertEquals, the thing being tested comes second.  If there is an error message, the first thing appears as the "expected value"

Reply:
When using assert, the first parameter is `expected` and second parameter is `actual` right? In this case, we are testing that the MetricsReporter's client.id will match the correct (expected) client id from the client. The client is the one generating the client.id, and the `MetricsReporter` needs to receive it correctly.`producer.getClientId()` should be first and `MockMetricsReporter.clientId` should be second?

Reply:
Sorry, you're right.  The thing being tested should be second.